### ISSUE TITLE
feat: support variable axes

### ISSUE THREAD

**Issue by qwerzl** [unjs/unifont#33]
### 🆒 Your use case

Coming from nuxt/fonts#141.

The current unifont only supports wght and ital as variable axes, but font providers like Google offer control over way more variable axes than these two. We should also support them to leverage the full potential of fonts like [Recursive](https://www.recursive.design/).

### 🆕 The solution you'd like

- [x] #34
- [ ] For providers that have not yet provided this kind of fine-grained control (which sadly means all providers other than google as of now), we should provide a built in variable axis subsetter.

@danielroe What do you think?

**Comment by danielroe** [comment]
agreed! we can support arbitrary axes in the resolve options but need to consider how they could be normalised across providers (or mark them experimental until we have more than one provider supporting them)
